model/src/main/resources/reference.yml

```diff
@@ -210,8 +210,6 @@ ort:
         # Command line options that do not affect the ScanCode output.
         commandLineNonConfig: '--processes 4'
```

Comment:
> It anyway is better implemented as a general post-processing
of scan results from all scanners that decomposes the licenses if
required.

We recently had a discussion about the programmatic use of the scanners, and what kind of guarantees the interface should provide. To my understanding, the consensus was that the interface should remain strict and not make too much assumptions about further processing of the returned scan results.  Has there been any changes to that consensus which I missed? (I'm asking that because, IIUC this part of the commit message again assumes the "weakening" of the interface of the individual scanners to be the right way to go, which IMO contradicts to what we agreed upon. What do you think?

Reply:
> To my understanding, the consensus was that the interface should remain strict and not make too much assumptions about further processing of the returned scan results.

That was not my understanding. *My* understanding was that we agreed that any post-processing should become part of the `(Path)ScannerWrapper`'s API eventually. That is, basically just the `postProcessScanSummary()` function would be moved from `Scanner` to `(Path)ScannerWrapper`, and the latter would call it as part of a default `scanPath()` implementation after calling (a yet to be introduced) `scanPathInternal()`.

Ultimately, I have two goals:
- avoid code duplication by not requiring scanner implementations to repeatedly implement the same post-processing steps (also not via helper functions that we provide, but must be called manually)
- safe-guard against "bad" scanner implementations by not allowing them to override / skip any of the mandatory post-processing

But I'm completely open to implementing these two points differently than outlined above. As mentioned in an earlier discussion, I'm not a fan of having both `scanPath()` and `scanPathInternal()`. An alternative could be to have a low-level scanner interface that scanners are supposed to implement, but for *using* scanners another high-level interface that implements the post-processing after calling the low-level interface must be used.